misc/debug-adapter/modules/debug-adapter-core/src/main/java/org/ballerinalang/debugadapter/evaluation/engine/JvmInstanceMethod.java

```diff
@@ -57,8 +65,44 @@ public Value invoke() throws EvaluationException {
         } catch (EvaluationException e) {
             throw e;
         } catch (Exception e) {
-            throw new EvaluationException(String.format(EvaluationExceptionKind.FUNCTION_EXECUTION_ERROR
-                    .getString(), methodRef.name()));
+            throw new EvaluationException(String.format(EvaluationExceptionKind.FUNCTION_EXECUTION_ERROR.getString(),
+                    methodRef.name()));
+        }
+    }
+
+    @Override
+    protected List<Value> getMethodArgs(JvmMethod method) throws EvaluationException {
+        try {
+            if (argValues == null && argEvaluators == null) {
+                throw new EvaluationException(String.format(EvaluationExceptionKind.FUNCTION_EXECUTION_ERROR
+                        .getString(), methodRef.name()));
+            }
+            if (argValues != null) {
+                return argValues;
+            }
+            List<Value> argValueList = new ArrayList<>();
+            // Evaluates all function argument expressions at first.
+            for (Evaluator argEvaluator : argEvaluators) {
+                argValueList.add(argEvaluator.evaluate().getJdiValue());
+                // Assuming all the arguments are positional args.
+                argValueList.add(EvaluationUtils.make(context, true).getJdiValue());
+            }
+
+            List<Type> types = method.methodRef.argumentTypes();
+            // Removes injected arguments added during the jvm method gen phase.
+            for (int index = types.size() - 1; index >= 0; index -= 2) {
+                types.remove(index);
+            }
+
+            // Todo - IMPORTANT: Add remaining steps to validate and match named, defaultable and rest args
+            // Todo - verify
+            // Here we use the parent strand instance to execute the function invocation expression.
+            Value parentStrand = getParentStrand();
+            argValueList.add(0, parentStrand);
+            return argValueList;
+        } catch (ClassNotLoadedException e) {
+            throw new EvaluationException(String.format(EvaluationExceptionKind.FUNCTION_EXECUTION_ERROR.getString(),
+                    methodRef.name()));
```

Comment:
`getMethodArgs(JvmMethod method)` method is almost repeated in `JvmStaticMethod` and `JvmInstanceMethod`. Shall we add them to a Util class and use it from there?

Reply:
What happens inside the `getMethodArgs()` depends on the method type and therefore I've declared it as abstract in the parent class(`org.ballerinalang.debugadapter.evaluation.engine.JvmMethod`). WDYT?

Reply:
+1 for declaring it as abstract in the parent class.